### PR TITLE
Course eval links

### PR DESCRIPTION
**Pull request by alex-shen1**
I've created a button that shows up on every class page that will link to the UVA student evaluation page for that particular class. 

## Code changes
- Added `eval_link()` to `models.Course`, which appends department mnemonic/number to standard student evaluation link
- Call `eval_link()` in a button on `course.html` to link to course evals
- Wrote a basic test for it (but the link it generates is worthless because the test model doesn't correspond to a real class)

### A note on aesthetics:

Right now, it just looks like this:

image

I'm aware that it looks terrible *and* if you examine the HTML code itself you can see that the button itself is literally just a wrapper for a hyperlink, but I submitted a mockup request with the design team and figured since I'm going to have to change it to look better anyways, it was more important to just get the basic functionality working.

### CODE REVIEW

File: tcf_website/models/models.py [hunk 1]
```diff
@@ -286,6 +286,15 @@ def code(self):
...
         return f"{self.subdepartment.mnemonic} {self.number}"
 
+    def eval_link(self):
+        """Returns link to student eval page for that class"""
+        link = "https://evals.itc.virginia.edu/course-selectionguide/pages/SGMain.jsp?cmp="
+        link += self.subdepartment.mnemonic
+        link += ","
+        link += str(self.number)
+        return link
+        # return "test"
```
Comment (saikonuri) [inline tcf_website/models/models.py:296]: Remove comment!

Comment (alex-shen1) [reply tcf_website/models/models.py:296]: I thought I removed it in one of the commits...must've accidentally not added it to the commit, will fix ASAP

File: tcf_website/templates/course/course.html [hunk 2]
```diff
@@ -17,6 +17,7 @@
             <h1 class="mr-md-3">{{ course.code }}</h1>
             <h2>{{ course.title }}</h2>
         </div>
+        <button type="button" ><a href={{course.eval_link}}>Student Evaluations (Requires NetBadge login)</a></button>
```
Comment (saikonuri) [inline tcf_website/templates/course/course.html:20]: Maybe make this a boostrap button for now? Unless you want to wait until you hear from UX people

Comment (alex-shen1) [reply tcf_website/templates/course/course.html:20]: Yeah I don't have a lot of experience w/ Bootstrap so I figured I'd wait to sort that out until I know exactly what the target is

**Comment by saikonuri** [comment]
Looks good Alex! I'll wait to approve it until you hear back from UX

File: tcf_website/models/models.py (hunk 1 above)
Comment (alex-shen1) [reply tcf_website/models/models.py]: Turns out I didn't hit the save button in VSCode after I deleted the line, should be fixed now

**Comment by j-alicia-long** [comment]
Hey Alex! since UX seem to have their hands full I can take a crack at this sometime tmrw. Right now I'm thinking an inline-button with the heading that says "UVA Evaluations" with a hover tooltip that says "Requires Netbadge Login"

File: tcf_website/models/models.py [hunk 3]
```diff
@@ -286,6 +286,14 @@ def code(self):
         """Returns the courses code string."""
         return f"{self.subdepartment.mnemonic} {self.number}"
 
+    def eval_link(self):
+        """Returns link to student eval page for that class"""
+        link = "https://evals.itc.virginia.edu/course-selectionguide/pages/SGMain.jsp?cmp="
+        link += self.subdepartment.mnemonic
```
Comment (j-alicia-long) [inline tcf_website/models/models.py:292]:
```suggestion
        link += self.subdepartment.mnemonic + "," + str(self.number)
```

Comment (j-alicia-long) [reply tcf_website/models/models.py:292]: Even better - use f-strings: `f"{self.subdepartment.mnemonic},{self.number}"`

File: tcf_website/templates/course/course.html (hunk 2 above)
Comment (j-alicia-long) [inline tcf_website/templates/course/course.html:20]: Would recommend you have the link open up in a new tab!

File: tcf_website/tests/test_course.py [hunk 4]
```diff
@@ -61,3 +61,11 @@ def test_average_difficulty_no_reviews(self):
         self.review2.delete()
 
         self.assertTrue(self.course.average_difficulty() is None)
+
+    def test_student_eval_link(self):
```
Comment (j-alicia-long) [inline tcf_website/tests/test_course.py:65]: BLESS you for writing new tests :') we gotta do more of that

**Review by j-alicia-long — CHANGES_REQUESTED** [review]
Gotta update UI & general code formatting, link opening in new tab, etc. Looks good overall!

File: tcf_website/templates/course/course.html (hunk 2 above)
Comment (alex-shen1) [reply tcf_website/templates/course/course.html]: Done!

**Comment by alex-shen1** [comment]
Looks like this now:
image

image

File: tcf_website/static/base/tcf-style.css [hunk 5]
```diff
@@ -76,7 +76,7 @@ background-color: var(--secondary-color);
 
 /* Accent color: Orange */
 .text-tcf-orange {
-color: var(--secondary-color);
+color: var(--accent-color);
```
Comment (j-alicia-long) [inline tcf_website/static/base/tcf-style.css:79]: thanks!

File: tcf_website/templates/course/course.html [hunk 6]
```diff
@@ -2,42 +2,64 @@
...
 {% block styles %}
-<link rel="stylesheet" href="{% static 'course/course.css' %}" />
-<link rel="stylesheet" href="{% static 'common/rating_card.css' %}" />
-<link rel="stylesheet" href="{% static 'common/show-hide.css' %}" />
-<link rel="stylesheet" href="{% static 'common/toolbar.css' %}" />
+    <link rel="stylesheet" href="{% static 'course/course.css' %}"/>
+    <link rel="stylesheet" href="{% static 'common/rating_card.css' %}"/>
+    <link rel="stylesheet" href="{% static 'common/show-hide.css' %}"/>
+    <link rel="stylesheet" href="{% static 'common/toolbar.css' %}"/>
+    <link rel="stylesheet" href="{% static 'base/tcf-style.css' %}"/>
```
Comment (j-alicia-long) [inline tcf_website/templates/course/course.html:9]: oh you don't need to add tcf-style.css to this template - it's already imported in base (and course.html extends from base). I'd remove it!

File: tcf_website/templates/course/course.html [hunk 7]
```diff
@@ -2,42 +2,64 @@
...
+            </ul>
+        </div>
+    </div>
+{% endblock %}
 
 {% block js %}
-<script type="module" src="{% static 'common/bind-show-hide.js' %}"></script>
+    <script type="module" src="{% static 'common/bind-show-hide.js' %}"></script>
+
+    {#initializes Bootstrap tooltip#}
```
Comment (j-alicia-long) [inline tcf_website/templates/course/course.html:59]: huh i didn't know html had this comment syntax

**Review by j-alicia-long — COMMENTED** [review]
looks good! just remove that one import and we're good to go

File: tcf_website/templates/course/course.html (hunk 7 above)
Comment (alex-shen1) [reply tcf_website/templates/course/course.html:59]: I didn't either! Ctrl+/ did though, it knows everything.

**Review by j-alicia-long — APPROVED** [review]
looks good!

**Pull request merged**